sympy/polys/factortools.py

```diff
@@ -323,7 +323,7 @@ def dup_zz_zassenhaus(f, K):
     fc = f[-1]
     A = dup_max_norm(f, K)
     b = dup_LC(f, K)
-    B = int(abs(K.sqrt(K(n + 1))*2**n*A*b))
+    B = int(dup_zz_mignotte_bound(f, K))
```

Comment:
It looks like the improved Mignotte bound cannot be used here. There is a test https://github.com/sympy/sympy/blob/0515aeec9ff25f5abb831f0d54a6eb6704084551/sympy/polys/factortools.py#L402
that fails if `B` is too small. I haven't seen the original documentation but it seems that this `B` should not be changed.